Session08/binarytree/solutions/BinarySearchTree.java

```diff
@@ -0,0 +1,123 @@
+// This class fails to compile, can you figure out why, and fix it?
+// This is beacuse BinaryTree has a constructor which is not default. Hence, when we extend this class and try to call a no-argument constructor, it'll give a compile-time error. so, we call it by passing the parameters (to instantiate the root node) in super() called inside the constructor of BinarySearchTree
+
+class BinarySearchTree extends BinaryTree {
+
+    private int flag = 0;
```

Comment:
Any reason why this is not a `boolean`? Also, why is it a member variable? What does "BinarySearchTree.flag" mean for the tree as a whole?